packages/admin-panel/src/VizBuilderApp/components/DataLibrary/AggregationDataLibrary.js

```diff
@@ -17,24 +17,43 @@ const aggregateToValue = aggregate =>
 
 const valueToAggregate = value => value.map(({ code, config }) => ({ type: code, config }));
 
-export const AggregationDataLibrary = ({ aggregate, onAggregateChange }) => {
+export const AggregationDataLibrary = ({ aggregate, onAggregateChange, onInvalidChange }) => {
   const [inputValue, setInputValue] = useState('');
-  const debouncedInputValue = useDebounce(inputValue, 200);
   const value = aggregateToValue(aggregate);
-  const { data: aggregationOptionSearchResults, isFetching } = useSearchAggregationOptions({
-    search: debouncedInputValue,
-  });
-  const options = inputValue ? aggregationOptionSearchResults : [];
+  const { data: options, isFetching } = useSearchAggregationOptions();
+
+  const onChange = (event, newValue) => onAggregateChange(valueToAggregate(newValue));
+  const onRemove = (event, option) => {
+    onChange(
+      event,
+      value.filter(item => option.code !== item.code),
+    );
+  };
 
   return (
     <DataLibrary
-      options={options}
+      options={options || []}
       value={value}
-      onChange={(event, newValue) => onAggregateChange(valueToAggregate(newValue))}
+      onChange={onChange}
       inputValue={inputValue}
       onInputChange={(event, newInputValue) => (event ? setInputValue(newInputValue) : false)}
       isLoading={isFetching}
       searchPageSize={MAX_RESULTS}
+      onMouseEnter={prefetchAggregationOptions}
+      optionComponent={(option, index, setEdittingOption) => (
```

Comment:
Instead of adding a config for option component (e.g: `isUsingJsonEditor`) to achieve requirement, personally I slightly incline to dynamically pass a react component as a function to `DataLibrary`. This can easier to tell the difference between `AggregationDataLibrary` and `DataElementLibrary` and easier to extend for `tranform` tab.  

It is called in https://github.com/beyondessential/tupaia/blob/7332677fa163f03ee2c3a177de1d17544dde74d0/packages/ui-components/src/components/DataLibrary/options.js#L200 

The downside is that further props changes need to be applied at both side.

Reply:
Yeah it's a little awkward that the props need to be specified on both ends...

You could perhaps work with a `additionalProps` object, and spread it here if you wanted? ie.
`(option, index, additionalProps) => ( <SelectedOptionWithJsonEditor  {...additionalProps} option={option} ... />`

But I think for now it's fine 👍